packages/synapse-react-client/src/components/useImmutableTableQuery.ts

```diff
@@ -159,24 +160,13 @@ export default function useImmutableTableQuery(
           ? queryRequest(getLastQueryRequest())
           : queryRequest,
       )
-      setLastQueryRequest(newQueryRequest)
 
-      if (newQueryRequest.query) {
-        const clonedQueryJson = JSON.stringify(newQueryRequest.query)
-        if (shouldDeepLink) {
-          const encodedQuery = encodeURIComponent(clonedQueryJson)
-          DeepLinkingUtils.updateUrlWithNewSearchParam(
-            'QueryWrapper',
-            componentIndex,
-            encodedQuery,
-          )
-        }
-        if (onQueryChange) {
-          onQueryChange(clonedQueryJson)
-        }
```

Comment:
onQueryChange code moved to setState handler above, deep linking code moved to its own effect below.